### PR TITLE
Limit Spring Integration message channel executor queues

### PR DESCRIPTION
**Pull request by kasarit**
The executors are now declared as beans so that the spring context manages their lifecycle.

### CODE REVIEW

File: src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java [hunk 1]
```diff
@@ -1,68 +1,129 @@
...
-    public MessageChannel finishChannel() {
-        return new PublishSubscribeChannel(Executors.newSingleThreadExecutor(newThreadFactory("Finish-")));
+    public MessageChannel errorLoggingChannel() {
+        return new PublishSubscribeChannel(errorLoggingExecutor());
+    }
+
+    private ExecutorService newBlockingSingleThreadExecutor(final String threadNamePrefix) {
+        final BlockingQueue<Runnable> queue = new ArrayBlockingQueue<>(executorQueueSize);
+        return new ThreadPoolExecutor(1, 1, 0, TimeUnit.SECONDS, queue,
+                newThreadFactory(threadNamePrefix), new CallerBlocksPolicy(Long.MAX_VALUE));
```
Comment (petringo) [inline src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java:126]: A shortcoming in the original plan... Now if the whole chain gets stuck for some unknown reason (other than database not responding, for which we are prepared), we get no clear signal of it.

To detect these situations, I'd suggest something like following (you might choose to create a new issue of it):

1. Register each phase executor in a health indicator component (one indicator for all executors).
2. Declare a configurable timeout with a decent default value for blocking.
3. When the timeout is reached, flag executor as unhealthy.
4. When blocking finishes, flag executor as healthy.
5. When all executors are flagged unhealthy, flag application state as unhealthy.

**Review by petringo — CHANGES_REQUESTED** [review]

File: src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java [hunk 2]
```diff
@@ -1,68 +1,129 @@
...
 @Configuration
 public class ChannelConfig {
 
+    @Value("${executor.queue-size}")
+    private int executorQueueSize;
+
     @Autowired
     private ThreadGroup aviationMessageArchiverThreadGroup;
 
+    @Bean
```
[the remark this answers is not in the document]
Comment (kasarit) [reply src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java:23]: No, Spring infers it automagically.

File: src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java (hunk 1 above)
Comment (kasarit) [reply src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java:126]: 👍 Created issue #70.

File: src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (petringo) [reply src/main/java/fi/fmi/avi/archiver/config/ChannelConfig.java:23]: You're right! I should've read the documentation.

**Review by petringo — APPROVED** [review]

**Pull request merged**